typespec-extension/src/code-model-builder.ts

```diff
@@ -519,18 +518,19 @@ export class CodeModelBuilder {
           apiVersion.version = version.value;
           codeModelClient.apiVersions.push(apiVersion);
         }
-      } else {
-        // fallback to @service.version
-        const service = getService(this.program, client.service);
-        if (service?.version) {
-          codeModelClient.apiVersions = [];
-          const apiVersion = new ApiVersion();
-          apiVersion.version = service.version;
-          codeModelClient.apiVersions.push(apiVersion);
-        } else {
-          throw new Error(`API version not available for client ${client.name}.`);
-        }
       }
+      // } else {
+      //   // fallback to @service.version
+      //   const service = getService(this.program, client.service);
+      //   if (service?.version) {
+      //     codeModelClient.apiVersions = [];
+      //     const apiVersion = new ApiVersion();
+      //     apiVersion.version = service.version;
+      //     codeModelClient.apiVersions.push(apiVersion);
+      //   } else {
+      //     throw new Error(`API version not available for client ${client.name}.`);
+      //   }
+      // }
```

Comment:
Do not use `@service.version` as api-version

Reply:
However, we may reconsider this

From Timothee

- if the service provides no versioning: then this is kinda an error if we were explicit about the param. Should just be a regular param
- if the service provide a version via `@service version` it technically is a constant param
- if the service provide `@versioned` then it should allow any of those values 

And I kind of agree. `@service` without `version` seems best suit to say that the service does not have any version support.